project_generator/project.py

```diff
@@ -525,52 +534,66 @@ def copy_files(self, proj_dic, tool):
         shutil.copy2(os.path.join(os.getcwd(), linker), os.path.join(os.getcwd(), proj_dic['output_dir']['path'], linker))
 
     @staticmethod
-    def scrape_dir(root, directory, project_name, board, list_sources):
-        data = {
-            'common': {
-                'linker_file': [],
-                'sources': [],
-                'includes': [],
-                'source_files_obj' : [],
-                'source_files_lib' : [],
-                'target': [],
-            }
-        }
-
-        linker_filetypes = FILES_EXTENSIONS['linker_file']
-        source_filetypes = FILES_EXTENSIONS['source_files_c'] + FILES_EXTENSIONS['source_files_cpp'] + FILES_EXTENSIONS['source_files_s']
-        include_filetypes = FILES_EXTENSIONS['include_paths']
-        lib_filetypes = FILES_EXTENSIONS['source_files_lib']
-        obj_filetypes = FILES_EXTENSIONS['source_files_obj']
+    def determine_tool(linker_ext):
+        if "sct" in linker_ext or "lin" in linker_ext:
+            return "uvision"
+        elif "ld" in linker_ext:
+            return "make_gcc_arm"
+        elif "icf" in linker_ext:
+            return "IAR"
```

Comment:
should we use lower case for tools?
It  would be return "iar". For ld should return gcc. This means that this is not tool as pgen understand it - isn't  this toolchain? Becuase tool can be make_gcc_arm or eclipse_gcc_project or uvision_gcc.